src/common/recordbatch/src/filter.rs

```diff
@@ -66,16 +66,19 @@ impl SimpleFilterEvaluator {
                         op = op.swap().unwrap();
                         (col, lit)
                     }
-                    _ => return None,
+                    _ => return Ok(None),
                 };
 
-                Some(Self {
+                let literal = rhs
+                    .to_scalar()
+                    .with_context(|_| ToArrowScalarSnafu { v: rhs.clone() })?;
```

Comment:
We can return `None` as `SimpleFilterEvaluator::try_new()` only returns `Some` when the expr can be converted into the filter.